packages/office-ui-fabric-react/src/components/Check/Check.base.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  classNamesFunction,
+  customizable,
+} from '@uifabric/utilities';
+import {
+  ICheckProps,
+  ICheckStyleProps,
+  ICheckStyles
+} from './Check.types';
+import { Icon } from '../../Icon';
+import { getStyles } from './Check.styles';
+
+const getClassNames = classNamesFunction<ICheckStyleProps, ICheckStyles>();
+
+@customizable('Check', ['theme'])
+export class CheckBase extends BaseComponent<ICheckProps, {}> {
+  public static defaultProps: ICheckProps = {
+    checked: false
+  };
+
+  public shouldComponentUpdate(newProps: ICheckProps) {
+    return this.props.checked !== newProps.checked;
+  }
+
+  public render(): JSX.Element {
+    const {
+      checked,
+      className,
+      theme
+    } = this.props;
+
+    const classNames = getClassNames(getStyles!, { theme: theme!, className, checked });
+
+    return (
+      <div
+        className={ classNames.root }
+      >
+        { Icon({
```

Comment:
I'm confused why this isn't JSX.

```tsx
<Icon className={ ... } iconName={ ... } />
```

Reply:
That was how it was written before, I was trying to keep the number of logic changes unrelated to mergeStyles to a minimum.